### PR TITLE
feat: Add template designs to auth page as per Spree 4.3

### PR DESCRIPTION
**Pull request by hamzanaveed9**
asset integration and auth-pages design in 4.3

### CODE REVIEW

File: app/views/spree/home/index.html.erb [hunk 1]
```diff
@@ -1,22 +0,0 @@
-<% if @homepage %>
```
Comment (yaseeniqbal575) [inline app/views/spree/home/index.html.erb]: why did you remove this file. Default file should be present

File: app/views/spree/shared/_header.html.erb [hunk 2]
```diff
@@ -1,28 +1,38 @@
...
-          <%= render partial: 'spree/shared/mobile_navigation' %>
+          <%#= render partial: 'spree/shared/mobile_navigation' %>
         </div>
       </div>
     </header>
+  </div> -->
+<%# end %>
+<% cache spree_menu_cache_key('header') do %>
+  <div id="spree-header">
+    <header id="header" >
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_header.html.erb]: We dnt need this header tag . As we dnt have this tag in out template. Please verify the header structure again.

File: app/views/spree/shared/_login.html.erb [hunk 3]
```diff
@@ -1,36 +1,63 @@
...
+    <%#= f.submit Spree.t(:login), class: 'btn btn-primary btn-block spree-btn mt-2' %>
+  <%# end %>
+  <%# if spree.respond_to?(:recover_password_path) %>
     <div class="text-center forgot-password mb-4">
-      <%= link_to Spree.t(:forgot_password), spree.recover_password_path %>
+      <%#= link_to Spree.t(:forgot_password), spree.recover_password_path %>
+    </div>
+  <%# end %>
+</div> -->
+<%= form_for Spree::User.new, as: :spree_user, url: spree.create_new_session_path do |f| %>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_login.html.erb]: This line of code is not matching from spree default spree.

The line is something like this in Default Spree: <%#= form_for Spree.user_class.new, as: :spree_user, url: spree_login_path do |f| %>

File: app/views/spree/home/index.html.erb (hunk 1 above)
Comment (hamzanaveed9) [reply app/views/spree/home/index.html.erb]: sir i removed it temporarily because i was putting static html in it i thought i will retrieve it once i will start working on index page

File: app/views/spree/shared/_head.html.erb [hunk 4]
```diff
@@ -10,4 +10,5 @@
 <%= render 'spree/shared/paths' %>
 <%= render 'spree/shared/locale_and_currency' %>
 <%= javascript_include_tag 'spree/frontend/all', defer: true, 'data-turbolinks-track': 'reload' %>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
```
Comment (om-ma) [inline app/views/spree/shared/_head.html.erb:13]: write down in you notes things to be pending

File: app/views/spree/user_mailer/confirmation_instructions.text.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Welcome <%= @email %>!
+
```
Comment (om-ma) [inline app/views/spree/user_mailer/confirmation_instructions.text.erb]: remove this file

File: app/views/spree/user_mailer/reset_password_instructions.text.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+A request to reset your password has been made.
+If you did not make this request, simply ignore this email.
```
Comment (om-ma) [inline app/views/spree/user_mailer/reset_password_instructions.text.erb]: remove this file

File: app/views/spree/shared/_login_bar.html.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+<script>if(typeof Spree.fetch_account !== 'undefined') Spree.fetch_account()</script>
```
Comment (om-ma) [inline app/views/spree/shared/_login_bar.html.erb]: verify it i think we remove it

File: app/views/spree/user_mailer/confirmation_instructions.text.erb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+Welcome <%= @email %>!
```
Comment (umairrazam) [inline app/views/spree/user_mailer/confirmation_instructions.text.erb:1]: remove it

File: app/views/spree/user_passwords/edit.html.erb [hunk 9]
```diff
@@ -1,4 +1,4 @@
- <%#= render :partial => 'spree/shared/error_messages', :locals => { :target => @spree_user } %>
+<%#= render :partial => 'spree/shared/error_messages', :locals => { :target => @spree_user } %>
 <!-- 
 <div class="container user-passwords">
```
Comment (om-ma) [inline app/views/spree/user_passwords/edit.html.erb:3]: create partial of deals blocks

File: app/views/spree/user_passwords/new.html.erb [hunk 10]
```diff
@@ -1,4 +1,3 @@
-
 <!-- <%#= render :partial => 'spree/shared/error_messages', :locals => { :target => @spree_user } %>
```
Comment (om-ma) [inline app/views/spree/user_passwords/new.html.erb:1]: create partial of deals blog

File: app/views/spree/user_registrations/new.html.erb [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+                         Already Registered?
+                        <% end%>
+                     </div>
+                     <div data-hook="login_extras"></div>
+                  </div>
+               <%end%>
+            </div>
+         </div>
+         <div class="col-lg-6">
+            <div class="latest-deals">
```
Comment (om-ma) [inline app/views/spree/user_registrations/new.html.erb]: create deals partial

File: app/views/spree/user_sessions/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+              </div>
+              <div class="products-wrapper d-sm-flex align-items-start">
+                 <div class="product-wrap">
+                    <a href="#">
+                       <div class="product-img">
+                          <%= image_tag "product-thumb-1.jpg" %>
+                          <div class="price"><span>$24.51</span>$13.51</div>
+                       </div>
+                       <div class="product-desc">
+                          <h4>Acorn Flavored Powder, Natural</h4>
```
Comment (om-ma) [inline app/views/spree/user_sessions/new.html.erb]: create deals partial

File: vendor/assets/javascripts/spree/frontend/all.js [hunk 13]
```diff
@@ -6,7 +6,5 @@
 //
 //= require jquery
 //= require jquery_ujs
-//= require accounting.min
```
Comment (om-ma) [inline vendor/assets/javascripts/spree/frontend/all.js:9]: why we remove this

Comment (hamzanaveed9) [reply vendor/assets/javascripts/spree/frontend/all.js:9]: it was not needed as suggested by yasin bhai so he deleted it

File: app/views/spree/shared/_upper_footer.html.erb [hunk 14]
```diff
@@ -0,0 +1,97 @@
+<div class="footer-links-wrap">
+  <div class="row no-gutters">
+     <div class="col-xl-2 tablet-hide"></div>
+     <div class="col-xl-2 col-lg-4">
+        <div class="footer-logo-wrap">
+           <figure>  <%= image_tag 'footer-logo.svg', alt: "Nature's Flavors" %></figure>
+           <p>Family owned and run for over 40 years. It is our mission to spread the Natural and Organic bounty of nature through our products.</p>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_upper_footer.html.erb:7]: We need to make this description dynamic as per Spree default

<%# if current_store.description %>
    <div class="d-none d-xl-block pt-xl-2 footer-spree-info">
      <%#= current_store.description %>
    </div>
  <%# end %>
  
  Adjust this code as per template

File: app/views/spree/shared/_upper_footer.html.erb [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+        </div>
+        <ul class="footer-nav">
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_upper_footer.html.erb:9]: Make the footer links dynamic as per Spree Default.

<%# if spree_menu('footer').present? %>
          <%# spree_menu('footer').children.each do |parent| %>
            <div class="col-3 mb-5">
              <div class="footer-spree-label">
                <%#if parent.container? %>
                  <%#= parent.name %>
                <%# else %>
                  <%#= spree_nav_link_tag(parent) %>
                <%# end %>
              </div>
              <%# parent.children.each do |child| %>
                <div class="pt-2">
                  <%#= spree_nav_link_tag(child) %>
                </div>
              <%# end %>
            </div>
          <%# end %>
        <%# end %>
        
        Please adjust this code inside template to print the links dynamic

File: app/views/spree/shared/_upper_footer.html.erb [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+                 <li><a href="#" class="facebook-icon">Facebook</a></li>
+                 <li><a href="#" class="instagram-icon">Intagram</a></li>
+                 <li><a href="#" class="pinterest-icon">Pinterest</a></li>
+                 <li><a href="#" class="youtube-icon">Youtube</a></li>
+              </ul>
+           </div>
+           <div class="col-lg-4 mobile-hide">
+              <div class="footer-address">
+                 <h5>Address</h5>
+                 <p>833 N. Elm Street, Orange,<br>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_upper_footer.html.erb:77]: Make the address info dynamic as per Spree

<%# if current_store.address %>
                <%#= current_store.address.html_safe %>
              <%# end %>

File: app/views/spree/shared/_upper_footer.html.erb [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+                    California 92867
+                 </p>
+              </div>
+              <div class="footer-phone mobile-hide"><a href="tel:1-714-744-3700">1-714-744-3700</a></div>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_upper_footer.html.erb:81]: Make the phone number information dynamic as per Spree

<%= current_store.contact_phone %>

File: app/views/spree/shared/_bottom_footer.html.erb [hunk 18]
```diff
@@ -0,0 +1,15 @@
+<div class="footer-copyright">
+   <div class="row no-gutters">
+      <div class="col-xl-2 tablet-hide"></div>
+      <div class="col-xl-4 col-lg-6">
+         <p>Copyright &copy; 1990-2021 Natures Flavors, Inc. All rights reserved.</p>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_bottom_footer.html.erb:5]: Add this copyright information into the partial and call the partial here

<%#= render 'spree/shared/copyright' %>

File: app/views/spree/shared/_login.html.erb [hunk 19]
```diff
@@ -1,36 +1,69 @@
-<div class="col-lg-11 mx-auto">
+<!-- <div class="col-lg-11 mx-auto">
   <h3 class="spree-mb-large spree-mt-large spree-header"><%= Spree.t(:log_in_to_continue) %></h3>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_login.html.erb]: spree default code is not commented well

File: app/views/spree/shared/_login.html.erb [hunk 20]
```diff
@@ -1,36 +1,69 @@
...
     <div class="text-center forgot-password mb-4">
-      <%= link_to Spree.t(:forgot_password), spree.recover_password_path %>
+      <%#= link_to Spree.t(:forgot_password), spree.recover_password_path %>
     </div>
+  <%# end %>
+</div> -->
+
+
+<%= form_for Spree.user_class.new, as: :spree_user, url: spree_login_path do |f| %>
+  <nav class="breadcrumb-wrap d-flex flex-wrap align-items-center justify-content-between" id="password-credentials">
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_login.html.erb]: Please remove this id="password-credentials" from this <nav> . This shouldnt be here

File: app/views/spree/shared/_login.html.erb [hunk 21]
```diff
@@ -1,36 +1,69 @@
...
+    </ol>
+  </nav>
+  <h1 class="title">Sign In</h1>
+  <div class="signin-note">Welcome to our new site. If you had an account on our previous system, please reset your password to access the new system by click 
+    <%= link_to  spree.recover_password_path do%>
+      Forgot your password.
+    <%end%>
+    Contact us if you need further assistance.
+  </div>
+  <div class="form-group form-floating has-icon mb-20" >
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_login.html.erb]: Please add an empty div on email/password field and add this hook into that div
id="password-credentials"

File: app/views/spree/shared/_nav_bar.html.erb [hunk 22]
```diff
@@ -27,9 +27,90 @@
...
+  <%# end %>
   <li id="link-to-cart">
-    <%= render 'spree/shared/cart', class: 'd-inline-block cart-icon', size: 36, item_count: 0 %>
+    <%#= render 'spree/shared/cart', class: 'd-inline-block cart-icon', size: 36, item_count: 0 %>
   </li>
-  <%= render partial: 'spree/shared/internationalization_options' %>
-</ul>
+  <%#= render partial: 'spree/shared/internationalization_options' %>
+</ul> -->
+<%# i have missed  id="link-to-account" because of design issue %>
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_nav_bar.html.erb:36]: please discuss about this point with Umer. We need this div and id at any cost

File: app/views/spree/shared/_search.html.erb [hunk 23]
```diff
@@ -1,15 +1,25 @@
...
             classes: 'd-inline',
             width: 27,
             height: 27) %>
-  </button>
+  </button> -->
+<%# end %>
+<%= form_tag spree.products_path, method: :get , id:'global-search-button' do%>
+  <div class="header-search tablet-hide" id="search-dropdown">
+    <span class="search-custom-btn"></span>
+    <%= search_field_tag :keywords,
```
Comment (yaseeniqbal575) [inline app/views/spree/shared/_search.html.erb:19]: You missed these options from search_field. As its present in above default spree

autofocus: true,
'aria-label': Spree.t(:search),

File: vendor/assets/stylesheets/spree/frontend/global.css.scss [hunk 24]
```diff
@@ -2365,13 +2524,13 @@ a.btn.btn-block {
 	display: block;
 }
 
-.social-nav a.facebook-icon {
-	background-image: image-url("facebook.svg");
-}
+// .social-nav a.facebook-icon {
```
Comment (yaseeniqbal575) [inline vendor/assets/stylesheets/spree/frontend/global.css.scss]: please uncomment this line. if its creating problem then discuss it with Umer

File: vendor/assets/stylesheets/spree/frontend/global.css.scss [hunk 25]
```diff
@@ -2365,13 +2524,13 @@ a.btn.btn-block {
...
-	background-image: image-url("facebook.svg");
-}
+// .social-nav a.facebook-icon {
+// 	background-image: image-url("facebook.svg");
+// }
 
-.social-nav a.instagram-icon {
-	background-image: image-url("instagram.svg");
-}
+// .social-nav a.instagram-icon {
```
Comment (yaseeniqbal575) [inline vendor/assets/stylesheets/spree/frontend/global.css.scss]: please uncomment this line. if its creating problem then discuss it with Umer

File: app/views/spree/shared/_nav_bar.html.erb (hunk 22 above)
Comment (hamzanaveed9) [reply app/views/spree/shared/_nav_bar.html.erb:36]: We have a separate ticket for that. Will handle in next sprint by Yaseen
